### ISSUE TITLE
Certain columns are not getting highlighted.

### ISSUE THREAD

**Issue by Anant1008** [ghiscoding/Angular-Slickgrid#739]
There are some identity grid columns which is not suppose to be shown hence we do not show these columns, on row selection the entire row gets highlighted.
But, when we revert back or show hidden identity columns. The cell of hidden column does not change color as per row selection.

### Your Environment

| Software          | Version(s) |
| ----------------- | ---------- |
| Angular           | 9.1.0        | 
| Angular-Slickgrid | 2.25        | 
| TypeScript        | 3.8.3        | 
| Operating System  | x.y        | 
| NPM/Node/Yarn     | x.y        | 

### Expected Behavior
When we add back hidden columns, those added columns should also get highlighted. 

### Current Behavior
When we hide some columns and select the row and then add those hidden columns, those added columns are not getting highlighted.
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/74755428/115838881-be6e7a80-a437-11eb-8ec0-fcef5591cacc.png)
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/74755428/115838887-c0383e00-a437-11eb-95b4-d2652d3e0177.png)

**Comment by ghiscoding** [comment]
You might need to re-render the grid (via `invalidate`) after selecting a row (via `onSelectedRowsChanged`) or after showing the column from the column picker and/or the grid menu (via `onColumnsChanged`). While calling the grid invalidate, it will most probably close any open editors so I guess it would be ok to do that on any of the 2 proposed way of doing it.

```html
<angular-slickgrid 
     gridId="grid2"
     [columnDefinitions]="columnDefinitions" 
     [gridOptions]="gridOptions" 
     [dataset]="dataset"
     (sgOnSelectedRowsChanged)="handleOnSelectedRowsChanged($event.detail.eventData, $event.detail.args)">
</angular-slickgrid>
```
```ts
handleOnSelectedRowsChanged(e, args) {
   args.grid.invalidate();
}
```

or ColumnPicker/GridMenu

```ts
this.gridOptions = {
   columnPicker: {
     onColumnsChanged: (e: Event, args: any) => args.grid.invalidate()
   },
   gridMenu: {
     onColumnsChanged: (e: Event, args: any) => args.grid.invalidate()
   },
};
```

**Comment by ghiscoding** [comment]
Actually I just tried it and that doesn't work because SlickGrid is only setting the `selected` CSS class only on the visible columns in the `handleSelectedRangesChanged` event handler on this [line](https://github.com/6pac/SlickGrid/blob/master/slick.grid.js#L2778)

What does work however is this 
```ts
this.gridOptions = {
   columnPicker: {
     onColumnsChanged: (e: Event, args: any) => args.grid.setSelectedRows(args.grid.getSelectedRows())
   },
   gridMenu: {
     onColumnsChanged: (e: Event, args: any) => args.grid.setSelectedRows(args.grid.getSelectedRows())
   },
};
```
it's basically recalling the row selection (read it and rewrite it back) will call that `handleSelectedRangesChanged` again and will add the `selected` CSS on all cells including the column you just showed back

**Comment by ghiscoding** [comment]
This is now fixed and released under the new [version 2.28.2](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v2.28.2)

Cheers ⭐

**Comment by Anant1008** [comment]
Thank you so much
